### ISSUE TITLE
None Gradient, when multiplying tensor by constant

### ISSUE THREAD

**Issue by BodaSadalla98** [pytorch/pytorch#91851]
### 🐛 Describe the bug

tensor. grad gives `None` when initializing a tensor with `requires_grad=True`, and multiplying it with a constant value.

```python 

import torch
a = torch.tensor(5.0, requires_grad=True) * 0.1
b = torch.tensor(2.0, requires_grad=True)
c = a + b
c.backward()

print(a.grad, b.grad)
```
**output: None, tensor(1.)**

---------------------------------------------------

The same doesn't happen when we init the tensor without the `required_grad` parameter and then do it separately.

```python 
import torch
a = torch.tensor(5.0) * 0.1
a.requires_grad = True
b = torch.tensor(2.0, requires_grad=True)
c = a + b
c.backward()
print(a.grad, b.grad)
```
**output: tensor(1.) tensor(1.)**
 

### Versions

PyTorch version: 1.12.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.10.6 (main, Oct 24 2022, 16:07:47) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3070
Nvidia driver version: 470.161.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] torch==1.12.1
[conda] numpy                     1.23.4                   pypi_0    pypi
[conda] torch                     1.12.1                   pypi_0    pypi

**Comment by ngimel** [comment]
this is expected behavior, `a` is non-leaf tensor in this case, and thus doesn't retain gradient by default. You can call `a.retain_grad()` if you want to save a's gradient.
